### ISSUE TITLE
multi-line syntax for tags and links

### ISSUE THREAD

**Issue by blais** [beancount/beancount#99]
**[Original report](https://bitbucket.org/blais/beancount/issue/99) by Stefano Zacchiroli (Bitbucket: [zacchiro](https://bitbucket.org/zacchiro), GitHub: [zacchiro](https://github.com/zacchiro)).**

----------------------------------------

Currently, #tags and ^links are bound to be written in-line, on the first line of a transaction, like this:
```
  2015-05-30 ! "Cable Co" "Phone Bill" #tag1 ˆlink1
    id: "TW378743437"               ; Meta-data
    Expenses:Home:Phone  87.45 USD
    Assets:Checking                 ; You may leave one amount out
```
whereas metadata are written on multiple lines.

This is unfortunate, as it discourages using multiple tags and links, barring having to maintain very long lines in your journal.

It would be nice if tags and links could be written both on the first line of a transaction and also elsewhere in it, e.g., between the first line and the first posting like this:
```
  2015-05-30 ! "Cable Co" "Phone Bill" #tag1 ˆlink1
    #tag2 #tag3
    ^link2
    #tag4
    ^link3 ^link4 ^link5
    id: "TW378743437"               ; Meta-data
    Expenses:Home:Phone  87.45 USD
    Assets:Checking                 ; You may leave one amount out
```

Thanks for considering,

**Comment by blais** [comment]
**Original comment by Stefano Zacchiroli (Bitbucket: [zacchiro](https://bitbucket.org/zacchiro), GitHub: [zacchiro](https://github.com/zacchiro)).**

----------------------------------------

Heya, I was wondering if you got a chance to look into this.
I'm planning my Ledger-CLI → Beancount migration, and whether this is going to be supported or not impacts my output formatter for Beancount entries.

Of course it's totally fine if you didn't get a chance to implement it yet ☺, I would like just to know whether, as Beancount maintainer, you approve of this change or not.
If so, I might give it a try at proposing a patch, but I won't bother unless I know you welcome this grammar change.

With many thanks for Beancount!

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

No I haven't had a chance.
I set some specific goals a long time ago and am deferring nearly all small changes until these are complete. Lately I got a big piece in (the new booking algorithm by default).
Thanks for reporting the issue,

**Comment by blais** [comment]
**Original comment by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

Ethan Glasser-Camp proposed a fix as PR 74.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Merged in tbm13/beancount/issue99 (pull request #74)

Fixed #99: allow tags and links to be spread over several lines

Approved-by: Martin Blais <blais@furius.ca>

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #99: allow tags and links to be spread over several lines

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Merged in tbm13/beancount/issue99 (pull request #74)

Fixed #99: allow tags and links to be spread over several lines

Approved-by: Martin Blais <blais@furius.ca>
